### PR TITLE
3704 by Inlead: Update events with missing p2b data.

### PR DESCRIPTION
**Pull request by inlead**
https://platform.dandigbib.org/issues/3704

### CODE REVIEW

File: modules/ding_place2book/ding_place2book.module [hunk 1]
```diff
@@ -584,10 +532,9 @@ function ding_place2book_event_op($op, $node, $place2book_id = NULL) {
 
       // Catch update error by checking the return code.
       if ($req_result->status_message != 'Accepted') {
-        // @todo: This error handling is not appropriate for normal editors.
-        // Replace with drupal_set_message + watchdog.
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.module]: This `todo` still applies. The message displayed to the editor is not very friendly/understandable.

File: modules/ding_place2book/ding_place2book.module [hunk 2]
```diff
@@ -584,10 +532,9 @@ function ding_place2book_event_op($op, $node, $place2book_id = NULL) {
...
         $p2b_error = $req_result->headers['status'] . ' - ' . $req_result->status_message;
         drupal_set_message(t('UPDATE was NOT PERFORMED on Place2book. Place2book returned this error:') . $p2b_error, 'error', FALSE);
+        watchdog('error', 'DELETE was NOT PERFORMED on Place2book. Place2book returned this error: @p2b_error', array('@p2b_error' => $p2b_error), WATCHDOG_ERROR);
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.module]: The message sent to watchdog does not correspond to the message displayed to the user. One mentioned `UPDATE`. The other mentions `DELETE`. Judging by the `$op` it seems as if `UPDATE` is right.

File: modules/ding_place2book/ding_place2book.install [hunk 3]
```diff
@@ -87,23 +94,126 @@ function ding_place2book_uninstall() {
...
+  ));
+}
+
+/**
+ * Update events p2b data to be stored in new field.
+ */
+function ding_place2book_update_7003(&$sandbox) {
+  if (!isset($sandbox['processed'])) {
+    $sandbox['total'] = db_query("select count(nid) as count from {ding_place2book} p2b
+      where p2b.place2book_id <> -1")->fetchField();
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.install]: Can thr `DING_PLACE2BOOK_NO_ID` constant be used here?

File: modules/ding_place2book/ding_place2book.install [hunk 4]
```diff
@@ -87,23 +94,126 @@ function ding_place2book_uninstall() {
...
+
+    $sandbox['processed'] = 0;
+  }
+
+  if ($sandbox['total']) {
+    $events = db_query("select nid, place2book_id from {ding_place2book} p2b
+      where p2b.place2book_id <> -1 LIMIT {$sandbox['processed']}, 20")->fetchAll();
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.install]: As above.

**Comment by kasperg** [comment]
:+1: I appreciate the cleanup effort which has been put into this.

File: modules/ding_place2book/ding_place2book.module [hunk 5]
```diff
@@ -501,13 +501,12 @@ function ding_place2book_event_op($op, $node, $place2book_id = NULL) {
...
       if (empty($req_result->headers['event-id'])) {
-        // @todo: This error handling is not appropriate for normal editors.
-        // Replace with drupal_set_message + watchdog.
         $p2b_error = $req_result->headers['status'] . ' - ' . $req_result->status_message;
-        drupal_set_message(t('INSERT was NOT PERFORMED on Place2book. Place2book returned this error:') . $p2b_error, 'error', FALSE);
+        drupal_set_message(t('Something went wrong. Event could not be created on Place2book. Useful information administrators:') . $p2b_error, 'error', FALSE);
+        watchdog('error', 'INSERT was NOT PERFORMED on Place2book: @p2b_error', array('@p2b_error' => $p2b_error), WATCHDOG_ERROR);
       }
       else {
-        drupal_set_message(t('Event created on Place2book with remote ID %id', array('%id' => $req_result->headers['event-id'])));
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.module]: Why is the "Something went wrong." text added? As I read the code this branch does not relate to error handling?

File: modules/ding_place2book/ding_place2book.module [hunk 6]
```diff
@@ -501,13 +501,12 @@ function ding_place2book_event_op($op, $node, $place2book_id = NULL) {
 
       // Catch insert error by checking the event id.
       if (empty($req_result->headers['event-id'])) {
-        // @todo: This error handling is not appropriate for normal editors.
-        // Replace with drupal_set_message + watchdog.
         $p2b_error = $req_result->headers['status'] . ' - ' . $req_result->status_message;
-        drupal_set_message(t('INSERT was NOT PERFORMED on Place2book. Place2book returned this error:') . $p2b_error, 'error', FALSE);
+        drupal_set_message(t('Something went wrong. Event could not be created on Place2book. Useful information administrators:') . $p2b_error, 'error', FALSE);
+        watchdog('error', 'INSERT was NOT PERFORMED on Place2book: @p2b_error', array('@p2b_error' => $p2b_error), WATCHDOG_ERROR);
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.module]: The `watchdog` type should be the module name - not `error`.

File: modules/ding_place2book/ding_place2book.module [hunk 7]
```diff
@@ -501,13 +501,12 @@ function ding_place2book_event_op($op, $node, $place2book_id = NULL) {
 
       // Catch insert error by checking the event id.
       if (empty($req_result->headers['event-id'])) {
-        // @todo: This error handling is not appropriate for normal editors.
-        // Replace with drupal_set_message + watchdog.
         $p2b_error = $req_result->headers['status'] . ' - ' . $req_result->status_message;
-        drupal_set_message(t('INSERT was NOT PERFORMED on Place2book. Place2book returned this error:') . $p2b_error, 'error', FALSE);
+        drupal_set_message(t('Something went wrong. Event could not be created on Place2book. Useful information administrators:') . $p2b_error, 'error', FALSE);
```
Comment (kasperg) [inline modules/ding_place2book/ding_place2book.module]: Nitpicking: The text `Useful information administrators` is not very readable.

**Comment by guddo** [comment]
@kasperg Fixed.

**Pull request merged**